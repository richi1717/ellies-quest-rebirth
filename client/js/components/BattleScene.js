import React from 'react';
import PropTypes from 'prop-types';
import { normalBattleMusic } from './SoundEffects';

export default function BattleScene(props) {
  return (
    <div className={`${props.battleScene}-battle battle`}>
      {props.playMusic && <normalBattleMusic />}
      {props.children}
    </div>
  );
}

BattleScene.defaultProps = {
  battleScene: 'forest',
  playMusic: false
};

BattleScene.propTypes = {
  battleScene: PropTypes.string,
  playMusic: PropTypes.bool,
  children: PropTypes.node.isRequired
};
