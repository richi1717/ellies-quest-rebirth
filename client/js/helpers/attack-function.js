var tmpl = require('../src/template.js')
var hasBeenKilled = 0

function renderAttackMenu(monster, character) {
  $('span.battle-menu-turn + span.menu-attack').remove()
  console.log('attack')
  length = $('section.enemy-sprites').length
  console.log(length)
  i = 0
  monsters = []
  console.log(i, monsters)
  console.log(hasBeenKilled)

  while (i < length) {
    i++
    enemy = (i + hasBeenKilled)
    nameOf = $('section.enemy' + enemy).data('nameOf')
    console.log(nameOf)
    target = 'enemy' + enemy
    monsters.push({ name: nameOf, target: target })
  }
  var html = tmpl.attackMenu({
    monster: monsters,
    character: character
  })
  return html
}

module.exports = {
  renderAttackMenu: renderAttackMenu
}