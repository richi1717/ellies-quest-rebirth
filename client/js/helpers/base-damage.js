export default function (str, level) {
  return str + (Math.ceil((str + level) / 3.2) * Math.ceil((str * level) / 3.2));
}
