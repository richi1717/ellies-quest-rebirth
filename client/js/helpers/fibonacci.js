export default function fibonacci(startAtZero, times = 10) {
  const numerator = times - 2;
  const arr = startAtZero ? [0, 1] : [1, 1];
  let i = 0;
  while (i < numerator) {
    arr.push(arr[i] + arr[i + 1]);
    i++;
  }
  return arr;
}

