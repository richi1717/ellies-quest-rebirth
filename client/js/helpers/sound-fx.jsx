import React, { Component } from 'react';

export class EnemyAttackFX extends Component {
  render() {
    return (
      <audio
        src="/music/swipe.mp3"
        name="media"
        type="audio/mpeg"
        controls
        autoPlay
      />
    );
  }
}

export class HeroAttackFX extends Component {
  render() {
    return (
      <audio
        src="/music/sword-slash.mp3"
        name="media"
        type="audio/mp3"
        controls
        autoPlay
      />
    );
  }
}

export class NormalBattleMusic extends Component {
  render() {
    return (
      <audio
        src="/music/better-battle.m4a"
        name="media"
        type="audio/mpeg"
        controls
        autoPlay
      />
    );
  }
}

export class BattleVictoryMusic extends Component {
  render() {
    return (
      <audio
        src="/music/1-06 Victory Fanfare.mp3"
        name="media"
        type="audio/mpeg"
        controls
        autoPlay
      />
    );
  }
}

export class FireMagicFX extends Component {
  render() {
    return (
      <audio
        src="/music/fire1.mp3"
        name="media"
        type="audio/mpeg"
        controls
        autoPlay
      />
    );
  }
}

export class CureMagicFX extends Component {
  render() {
    return (
      <audio
        src="/music/healingSpell.mp3"
        name="media"
        type="audio/mpeg"
        controls
        autoPlay
      />
    );
  }
}

export class CursorMovementFX extends Component {
  render() {
    return (
      <audio
        src="/music/cursor.mp3"
        name="media"
        type="audio/mpeg"
        controls
        autoPlay
      />
    );
  }
}

export class LightningMagicFX extends Component {
  render() {
    return (
      <audio
        src="/music/lightning1.m4a"
        name="media"
        type="audio/mpeg"
        controls
        autoPlay
      />
    );
  }
}

export class BossBattleMusicPart1 extends Component {
  render() {
    return (
      <audio
        src="/music/boss-battle1.m4a"
        name="media"
        type="audio/mpeg"
        controls
        autoPlay
      />
    );
  }
}

export class BossBattleMusicPart2 extends Component {
  render() {
    return (
      <audio
        src="/music/boss-battle2.m4a"
        name="media"
        type="audio/mpeg"
        controls
        autoPlay
      />
    );
  }
}

export class GameOverMusic extends Component {
  render() {
    return (
      <audio
        src="/music/19 Game Over.mp3"
        name="media"
        type="audio/mpeg"
        controls
        autoPlay
      />
    );
  }
}

export class IntroMusic extends Component {
  render() {
    return (
      <audio
        src="/music/3-16 The Prelude.mp3"
        name="media"
        type="audio/mpeg"
        controls
        autoPlay
      />
    );
  }
}

export class OverworldMusic extends Component {
  render() {
    return (
      <audio
        src="/music/2-01 Terra's Theme.mp3"
        name="media"
        type="audio/mpeg"
        controls
        autoPlay
      />
    );
  }
}
