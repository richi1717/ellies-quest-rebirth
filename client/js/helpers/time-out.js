export function setTimeOutHelper(time, func, arg1, arg2) {
  return setTimeout(function () {
    func(arg1, arg2);
  }, time);
}
