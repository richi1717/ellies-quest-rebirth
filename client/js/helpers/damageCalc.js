import _random from 'lodash.random';
import { calcLevel } from '../helpers/levelCalc';

export function getBaseDamage(attacker) {
  const str = attacker.str;
  const lvl = attacker.level ? attacker.level : calcLevel(attacker.exp);

  return str + (((str + lvl) / 3.2) * ((str * lvl) / 3.2));
}

export default function damageCalc(attacker, defender) {
  const critical = _random(1, 100) % 26 === 0;
  const power = critical ? _random(2.9, 3.3) : _random(1.8, 2.7);
  const def = defender.def;
  const base = getBaseDamage(attacker);
  let dmg = Math.ceil((power * (512 - def) * base) / (1.6 * 512));
  dmg = dmg > 0 ? dmg : 1;

  return dmg;
}
