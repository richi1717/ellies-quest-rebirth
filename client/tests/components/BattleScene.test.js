import React from 'react';
import { shallow } from 'enzyme';
import BattleScene from '../../js/components/BattleScene';
import { normalBattleMusic } from '../../js/helpers/soundEffects';

jest.mock('../../js/helpers/soundEffects');

describe('<BattleScene />', () => {
  let battleScene;

  beforeEach(() => {
    battleScene = shallow(
      <BattleScene battleScene="grass" playMusic>
        <div className="test" />
      </BattleScene>
    );
  });

  test('should match snapshot', () => {
    expect(battleScene.getNode()).toMatchSnapshot();
  });

  test('should have default propTypes', () => {
    battleScene = shallow(
      <BattleScene>
        <div className="test" />
      </BattleScene>
    );

    expect(battleScene).toHaveClassName('forest-battle');
    expect(battleScene.find(normalBattleMusic)).toHaveLength(0);
  });

  test('should have a className of grass-battle battle', () => {
    expect(battleScene).toHaveClassName('grass-battle battle');
  });

  test('should call normalBattleMusic when playMusic is true', () => {
    expect(normalBattleMusic).toHaveBeenCalled();
  });

  test('should have a child with className of test', () => {
    expect(battleScene.find('div').at(1)).toHaveClassName('test');
  });
});

