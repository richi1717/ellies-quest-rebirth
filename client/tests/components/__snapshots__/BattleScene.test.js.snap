// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BattleScene /> should match snapshot 1`] = `
<div
  className="grass-battle battle"
>
  <div
    className="test"
  />
</div>
`;
